frontend/apps/reader/modules/readerdictionary.lua

```diff
@@ -1040,12 +1050,36 @@ end
 
 function ReaderDictionary:onSaveSettings()
     if self.ui.doc_settings then
-        logger.dbg("save default dictionary", self.default_dictionary)
-        self.ui.doc_settings:saveSetting("default_dictionary", self.default_dictionary)
+        self.ui.doc_settings:saveSetting("preferred_dictionaries", self.preferred_dictionaries)
         self.ui.doc_settings:saveSetting("disable_fuzzy_search", self.disable_fuzzy_search)
     end
 end
 
+function ReaderDictionary:onTogglePreferredDict(dict)
+    if not self.preferred_dictionaries then
+        -- Invoked from FileManager: no preferred dict to manage
+        return true
+    end
+    local removed = false
+    for idx, name in ipairs(self.preferred_dictionaries) do
+        if dict == name then
+            removed = true
+            table.remove(self.preferred_dictionaries, idx)
+            break
+        end
+    end
+    if not removed then -- insert it as first
+        table.insert(self.preferred_dictionaries, 1, dict)
+    end
+    UIManager:show(InfoMessage:new{
+        text = removed and T(_("%1 is no more a preferred dictionary for this document."), dict)
+                        or T(_("%1 is now the first preferred dictionary for this document."), dict),
```

Comment:
The `first` feels redundant here. Thoughts?

Reply:
For me, it hints/documents that when you tap twice on something that was (2), it will be re-added as being (1) - as you can have multiple preferred dicts. So, it always becomes the first preferred dict.

Reply:
most preferred?

First preferred definitely sounds a bit weird.

Reply:
"most preferred" sounds strange to my non-native-english ears :) 
But ok with it or just "preferred" (removing "first") if you feel it's fine and enough.

Reply:
For reference:

https://eur-lex.europa.eu/LexUriServ/LexUriServ.do?uri=COM:2010:0473:FIN:EN:PDF
> Home-made  explosives,  in  turn,  are  the  tool  most  preferred  by  terrorists  and  other  criminals  to  perpetrate  attacks.

https://eur-lex.europa.eu/LexUriServ/LexUriServ.do?uri=COM:2010:0473:FIN:FR:PDF
> Or  ce  sont  précisément  les  explosifs  de  cette  nature  que  les  terroristes  et  autres  criminels  préfèrent  utiliser  pour  perpétrer des attentats.

Reply:
> But ok with it or just "preferred" (removing "first") if you feel it's fine and enough.

Fine by me. ;-)